internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go

```diff
@@ -423,22 +486,119 @@ func (r MsSqlManagedInstanceResource) Update() sdk.ResourceFunc {
+				}
+
+				aadAdminRemoved := checkAADAdminRemoved(metadata.ResourceData)
+				if aadAdminRemoved && aadAdminExists {
+					future, err := azureADAuthenticationOnlyClient.Delete(ctx, id.ResourceGroup, id.Name)
+					if err != nil {
+						log.Printf("[INFO] Deletion of AAD Authentication Only failed for %s: %+v", *id, err)
+						return fmt.Errorf("disabling AAD Authentication Only for %s: %+v", *id, err)
+					}
+
+					if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+						return fmt.Errorf("waiting to disable AAD Authentication Only for %s: %+v", *id, err)
+					}
+
+					resp, err := adminClient.Delete(ctx, id.ResourceGroup, id.Name)
+					if err != nil {
+						return fmt.Errorf("removing the AAD Administrator for %s: %+v", *id, err)
+					}
+					if err = resp.WaitForCompletionRef(ctx, client.Client); err != nil {
+						return fmt.Errorf("waiting for removal of AAD Administrator for %s: %+v", *id, err)
+					}
+				}
+
+				aadAdminProps, err := expandMsSqlManagedInstanceAdministrators(state.AzureActiveDirectoryAdministrator)
+				if err != nil {
+					return err
+				}
+				if aadAdminProps != nil {
+					aadAdminChanged := checkAADAdminChanged(metadata.ResourceData)
+					if aadAdminChanged {
+						// if enabling AAD Authentication only, AAD admin must be set first.
+						future, err := adminClient.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, *aadAdminProps)
+						if err != nil {
+							return fmt.Errorf("creating AAD Administrator for %s: %+v", *id, err)
+						}
+						if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+							return fmt.Errorf("waiting for creation of AAD Administrator for %s: %+v", *id, err)
+						}
+
+					}
+					aadAuthOnlyChanged := checkAADAuthOnlyChanged(metadata.ResourceData)
+					if aadAuthOnlyChanged {
+						aadOnlyAuthenticationsProps := sql.ManagedInstanceAzureADOnlyAuthentication{
+							ManagedInstanceAzureADOnlyAuthProperties: &sql.ManagedInstanceAzureADOnlyAuthProperties{
+								AzureADOnlyAuthentication: pointer.To(expandMsSqlManagedInstanceAadAuthenticationOnly(state.AzureActiveDirectoryAdministrator)),
+							},
+						}
+						resp, err := azureADAuthenticationOnlyClient.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, aadOnlyAuthenticationsProps)
+						if err != nil {
+							return fmt.Errorf("setting `azuread_authentication_only_enabled` for %s: %+v", *id, err)
+						}
+						if err = resp.WaitForCompletionRef(ctx, client.Client); err != nil {
+							return fmt.Errorf("waiting to set `azuread_authentication_only_enabled` for  %s: %+v", *id, err)
+						}
+					}
+				}
+
+				administrators, err := expandMsSqlManagedInstanceExternalAdministrators(state.AzureActiveDirectoryAdministrator)
+				if err != nil {
+					return fmt.Errorf("expanding `azure_active_directory_administrator` for SQL Managed Instance Server %q: %v", id.ID(), err)
+				}
+				properties.ManagedInstanceProperties.Administrators = administrators
```

Comment:
(second instance of setting this, per previous comment)

Reply:
Setting the second instance of this is not necessary, I have removed this setting.